docs/stacks-guide-aws.md

```diff
@@ -0,0 +1,911 @@
+```
+<details> <summary>output</summary>
+
+```bash
+namespace/infra-aws created
+```
+</details>
+
+Now we can create the AWS stack by using Crossplane CLI.  Since this is an infrastructure stack, 
+we need to specify that it's cluster-scoped via `--cluster` flag.
+```bash
+kubectl crossplane stack generate-install --cluster 'crossplane/stack-aws:master' stack-aws | kubectl apply --namespace ${INFRA_NAMESPACE} -f -
+```
+
+The next steps assume that you installed AWS stack into the `infra-aws` namespace.
+
+## Configure the AWS account
+
+An [AWS user] with `Administrative` priviliges is needed to enable Crossplane to
+create the required resources. Once the user is provisioned, an [Access Key]
+need to be created to enable the user to have API access. Next, using this
+access key credentials, one needs to have [`aws` command line tool] [installed]
+and [configured]. Then, the credentials and configuration will reside in
+`~/.aws/credentials` and `~/.aws/config` respectively, which will be consumed in
+the next step.
+
+When configuring aws CLI, it is recommended that the user credentials are
+configured under a specific [aws named profile] other than `default`. In this
+guide we are assuming that the credentials are configured under
+`crossplane-user` profile, but you can use other profiles as well. Let's store
+the profile name in `aws_profile` environment variable to use later:
+
+```bash
+aws_profile=crossplane-user
+```
+
+## Configure Crossplane Provider for AWS
+
+Crossplane uses the aws user credentials that was configured in the previous
+step, to create resources in AWS. These credentials will be stored as a
+[secret], and is managed them by an  [`aw provider`] instance. In addition to
+the credentials, the AWS region is also read from the configuration to target a
+specific region.
+
+To store the credentials as a secret, run:
+```bash
+# retrieve profile's credentials, save it under 'default' profile, and base64 encode it
+AWS_CREDS_BASE64=$(cat ${HOME}/.aws/credentials | awk '/["$aws_profile"]/ {getline; print $0}' | awk 'NR==1{print "[default]"}1' | base64 | tr -d "\n")
+
+# retrieve the profile's region from config
+AWS_REGION=$(awk '/["$aws_profile"]/ {getline; print $3}' ${HOME}/.aws/config)
+```
+
+At this point, the region and the encoded credentials are stored in respective
+environment variables. Next, we'll need to create an instance of [`aw
+provider`]:
+
+```bash
+cat > ${WORK_DIR}/provider.yaml <<EOF
+---
+apiVersion: v1
+data:
+  credentials: ${AWS_CREDS_BASE64}
+kind: Secret
+metadata:
+  name: aws-user-creds
+  namespace: ${INFRA_NAMESPACE}
+type: Opaque
+---
+apiVersion: aws.crossplane.io/v1alpha2
+kind: Provider
+metadata:
+  name: aws-provider
+  namespace: ${INFRA_NAMESPACE}
+spec:
+  credentialsSecretRef:
+    key: credentials
+    name: aws-user-creds
+  region: ${AWS_REGION}
+EOF
+```
+
+Then, apply it to the cluster:
+
+```bash
+kubectl apply -f "${WORK_DIR}/provider.yaml"
```

Comment:
In other providers we made the decision to include the `kubectl apply` commands in the `cat` code block. Might be nice to have consistency here,